e2e/nomostest/reset.go

```diff
@@ -0,0 +1,531 @@
+			nt.T.Logf("[RESET] RepoSync %s managed by %q", rsNN, manager)
+			if !IsDeletionPropagationEnabled(rs) {
+				// If you go this error, make sure your test cleanup ensures
+				// that the managed RepoSync has deletion propagation enabled.
+				return errors.Errorf("RepoSync %s managed by %q does NOT have deletion propagation enabled: test reset incomplete", rsNN, manager)
+			}
+			continue
+		}
+
+		// Enable deletion propagation, if not enabled
+		if EnableDeletionPropagation(rs) {
+			nt.T.Logf("[RESET] Enabling deletion propagation on RepoSync %s", rsNN)
+			if err := nt.Update(rs); err != nil {
+				return err
+			}
+			WatchForObject(nt, kinds.RepoSyncV1Beta1(), rs.Name, rs.Namespace, []Predicate{
+				HasFinalizer(metadata.ReconcilerFinalizer),
+			}, WaitStrategy(WaitFailureStrategyError))
+		}
+
+		// Create RoleBinding, if not found (ensure finalizer has permission)
+		rsCRB := RepoSyncRoleBinding(rsNN)
+		if err := nt.Create(rsCRB); err != nil {
+			if !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+
+		// Print reconciler logs in case of failure.
+		// This ensures the logs are printed, even if the reconciler is deleted.
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		go TailReconcilerLogs(ctx, nt, NsReconcilerObjectKey(rsNN.Namespace, rsNN.Name))
+
+		// DeletePropagationBackground is required when deleting RSyncs with
+		// dependencies that have owners references. Otherwise the reconciler
+		// and dependenencies will be garbage collected before the finalizer
+		// can delete the managed resources.
+		// TODO: Remove explicit Background policy after the reconciler-manager finalizer is added.
+		nt.T.Logf("[RESET] Deleting RepoSync %s", rsNN)
+		if err := nt.Delete(rs, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
+			return err
+		}
+	}
+	for _, item := range rsList {
+		obj := &item
+		nn := client.ObjectKeyFromObject(obj)
+		nt.T.Logf("[RESET] Waiting for deletion of RepoSync %s ...", nn)
+		// TODO: Replace WaitForNotFound with something that returns an error
+		WaitForNotFound(nt, kinds.RepoSyncV1Beta1(), obj.Name, obj.Namespace,
+			WaitStrategy(WaitFailureStrategyError))
+	}
+
+	// Delete any RoleBindings left behind.
+	// For central control, the parent RSync _should_ handle deleting the RB,
+	// but for delegated control and other edge cases clean them up regardless.
+	nt.T.Log("[RESET] Deleting test RoleBindings")
+	var rbs []client.Object
+	for rsNN := range record.Objects {
+		rbs = append(rbs, RepoSyncRoleBinding(rsNN))
+	}
+	if err := batchDeleteAndWait(nt, rbs...); err != nil {
+		return err
+	}
+
+	// Delete any ClusterRoleBindings left behind.
+	// CRBs are usually only applied if PSP was enabled, but clean them up regardless.
+	nt.T.Log("[RESET] Deleting test ClusterRoleBindings")
+	var crbs []client.Object
+	for rsNN := range record.Objects {
+		crbs = append(crbs, repoSyncClusterRoleBinding(rsNN))
```

Comment:
hmm, I didn't realize the tests were already using a `ClusterRoleBinding` for RepoSyncs. We might want to limit this down to a `RoleBinding` to validate more realistic use case scenarios, but I don't view this change as a regression since we were already using `ClusterRoleBinding`.

Reply:
Both setup options use a CRB if PSP is enabled, because the RB will violate default policy which disallows cross-namespace permissions (granting reconciler in namespace A permission to read/write objects in namespace B).

Why they dont just modify the policy, i don't know....